### ISSUE TITLE
Confused about computing the estimated noise ratio ε

### ISSUE THREAD

**Issue by DanaIliescu** [chenpf1025/noisy_label_understanding_utilizing#5]
Firstly, thank you for your paper, I am trying to implement it in a language identification setup with asymmetric / real-word noisy labels. However, I have a hard time understanding how you compute the estimated noise ratio ε (step 8 in the pseudo-code of algorithm 2 in the paper, leading to Eq. 4). I have already looked at the paper and the code, but I am still confused about it. Would you be so kind to explain a bit how that is calculated?

**Comment by chenpf1025** [comment]
Thank you for your interest. In our implementation, we estimated the noise ratio by solving Eq. 4. We first obtain the left side of Eq. 4 (which is the test accuracy on a noisy set) through experiments, then we estimate the noise ratio by solving Eq. 4. Note that the test accuracy (on a noisy set) can be obtained by firstly training and then validating on noisy samples.

Theoretically, the estimation of Eq. 4 should work well when the noise is symmetric. While for our robust training algorithm, we found that it is not really sensitive to the estimated noise ratio.

**Comment by chengstark** [comment]
What is Num_top in the original algorithm in the paper?

**Comment by chenpf1025** [comment]
> What is Num_top in the original algorithm in the paper?

It is an empirical hyperparameter. To select more samples (but not to estimate \epsilon), we can select a sample if the top-k  prediction hits the label.

**Comment by chengstark** [comment]
Thank you!
